libraries/AP_HAL_Linux/SPIUARTDriver.cpp

```diff
@@ -49,31 +49,8 @@ void SPIUARTDriver::begin(uint32_t b, uint16_t rxS, uint16_t txS)
         txS = 2048;
     }
 
-    /*
-     allocate the read buffer
-   */
-    if (rxS != 0 && rxS != _readbuf_size) {
-        _readbuf_size = rxS;
-        if (_readbuf != NULL) {
-            free(_readbuf);
-        }
-        _readbuf = (uint8_t *)malloc(_readbuf_size);
-        _readbuf_head = 0;
-        _readbuf_tail = 0;
-    }
-
-    /*
-     allocate the write buffer
-   */
-    if (txS != 0 && txS != _writebuf_size) {
-        _writebuf_size = txS;
-        if (_writebuf != NULL) {
-            free(_writebuf);
-        }
-        _writebuf = (uint8_t *)malloc(_writebuf_size);
-        _writebuf_head = 0;
-        _writebuf_tail = 0;
-    }
+    _readbuf.set_size(rxS);
+    _writebuf.set_size(txS);
```

Comment:
it's never 0, see lines above.  And it doesn't change if the size is the same due to the implementation of set_size().